RecoTauTag/RecoTau/test/rerunMVAIsolationOnMiniAOD_Phase2.py

```diff
@@ -0,0 +1,136 @@
+process.rerunDiscriminationByIsolationMVAPhase2VTight.mapping[0].cut = cms.string("RecoTauTag_tauIdMVAIsoPhase2_WPEff50")
+process.rerunDiscriminationByIsolationMVAPhase2VVTight = process.rerunDiscriminationByIsolationMVAPhase2VVLoose.clone()
+process.rerunDiscriminationByIsolationMVAPhase2VVTight.mapping[0].cut = cms.string("RecoTauTag_tauIdMVAIsoPhase2_WPEff40")
+
+process.rerunMvaIsolation2Seq_Phase2 = cms.Sequence(
+    process.rerunDiscriminationByIsolationMVAPhase2raw
+    * process.rerunDiscriminationByIsolationMVAPhase2VVLoose
+    * process.rerunDiscriminationByIsolationMVAPhase2VLoose
+    * process.rerunDiscriminationByIsolationMVAPhase2Loose
+    * process.rerunDiscriminationByIsolationMVAPhase2Medium
+    * process.rerunDiscriminationByIsolationMVAPhase2Tight
+    * process.rerunDiscriminationByIsolationMVAPhase2VTight
+    * process.rerunDiscriminationByIsolationMVAPhase2VVTight
+)
+
+# embed new id's into tau
+embedID = cms.EDProducer("PATTauIDEmbedder",
+    src = cms.InputTag('slimmedTaus'),
+    tauIDSources = cms.PSet(
+        byIsolationMVAPhase2raw = cms.InputTag('rerunDiscriminationByIsolationMVAPhase2raw'),
+        byVVLooseIsolationMVAPhase2New = cms.InputTag('rerunDiscriminationByIsolationMVAPhase2VVLoose'),
+        byVLooseIsolationMVAPhase2New = cms.InputTag('rerunDiscriminationByIsolationMVAPhase2VLoose'),
+        byLooseIsolationMVAPhase2New = cms.InputTag('rerunDiscriminationByIsolationMVAPhase2Loose'),
+        byMediumIsolationMVAPhase2New = cms.InputTag('rerunDiscriminationByIsolationMVAPhase2Medium'),
+        byTightIsolationMVAPhase2New = cms.InputTag('rerunDiscriminationByIsolationMVAPhase2Tight'),
+        byVTightIsolationMVAPhase2New = cms.InputTag('rerunDiscriminationByIsolationMVAPhase2VTight'),
+        byVVTightIsolationMVAPhase2New = cms.InputTag('rerunDiscriminationByIsolationMVAPhase2VVTight'),
+    ),
+)
+setattr(process, "newTauIDsEmbedded", embedID)
+
+## added for mvaIsolation on miniAOD testing
+#process.out = cms.OutputModule("PoolOutputModule",
+#    fileName = cms.untracked.string(filetag+'_miniaod.root'),
+#    ## save only events passing the full path
+#    SelectEvents = cms.untracked.PSet( SelectEvents = cms.vstring('p') ),
+#    ## save PAT output; you need a '*' to unpack the list of commands
+#    ##'patEventContent'
+#    outputCommands = cms.untracked.vstring(
+#        'drop *',
+#        'keep *_newTauIDsEmbedded_*_*',
+#        'keep *_prunedGenParticles_*_*'
+#    )
+#)
+
+#process.genVisTauProducer = cms.EDProducer("GenVisTauProducer",
+#    genParticleCollection = cms.InputTag("prunedGenParticles")
+#)
+
+process.rerunMVAIsolationOnMiniAOD_Phase2 = cms.EDAnalyzer(
+    'rerunMVAIsolationOnMiniAOD_Phase2',
+    genJetCollection = cms.InputTag("slimmedGenJets"), #comment out to run on data
+    #genVisTauCollection = cms.InputTag("genVisTauProducer:genVisTaus"), #comment out if running on data
```

Comment:
Will it be useful to add `isData` switch in the analyser and this config to read or not this genVisTau collection? This can be also checked in runtime with `event.isRealData()`, where `event` is `edm::Event& event`.

Reply:
The way the code is setup is that the .cc file explicitly checks if the parameter was specified in the cfg, if it is not specified in the cfg then the analyzer .cc file does not load the collections:
https://github.com/cms-tau-pog/cmssw/blob/8c1791c61fce4b13fffae40cc83a2c75dae3469e/RecoTauTag/RecoTau/plugins/rerunMVAIsolationOnMiniAOD_Phase2.cc#L49-L63

The same concern would not only be for the genVisTau collection, but also for the slimmedGenJets and prunedGenParticles collections:
https://github.com/cms-tau-pog/cmssw/blob/8c1791c61fce4b13fffae40cc83a2c75dae3469e/RecoTauTag/RecoTau/test/rerunMVAIsolationOnMiniAOD_Phase2.py#L123-L125

With that said, it would be even safer to encapsulate the loading of these collections in the .cc file within a isRealData flag. I like this idea -- thank you.

The genVisTau collection is a collection I make with a separate producer which I did not include in this PR, that is why all references to it in this PR are commented out.
Within the existing tau code, is there an "official" producer which makes a collection of generator visible taus in which I could load?

Reply:
OK, it makes sense to me. Please see also my comment above on `TauGenJets`.